### PR TITLE
Removed 2 libraries with vulnerabilities

### PR DESCRIPTION
**Pull request by satsen**
The versions of Guava and Commons IO used in Appkit have vulnerabilities. Even if the vulnerable code might not be invoked (I didn't check), it's a bad look because all users of the library receive warnings. These libraries are not used much at all so removing them is the best solution.

### CODE REVIEW

**Review by kushti — APPROVED** [review]

File: common/src/test/scala/org/ergoplatform/appkit/SecretStorageSpec.scala [hunk 1]
```diff
@@ -69,7 +70,7 @@ class SecretStorageSpec extends AnyPropSpec with Matchers with ScalaCheckDrivenP
   }
 
   def withTempDir(block: File => Unit): Unit = {
-    val dir = Files.createTempDir()
+    val dir = Files.createTempDirectory("").toFile
```
Comment (aslesarenko) [inline common/src/test/scala/org/ergoplatform/appkit/SecretStorageSpec.scala:73]: Is this method available in Java 7?

File: lib-api/src/main/java/org/ergoplatform/appkit/BoxOperations.java [hunk 2]
```diff
@@ -63,8 +61,9 @@ public static BoxOperations createForSenders(List<Address> senders, BlockchainCo
      */
     public static BoxOperations createForEip3Prover(ErgoProver senderProver, BlockchainContext ctx) {
         List<Address> eip3Addresses = senderProver.getEip3Addresses();
-        checkState(eip3Addresses.size() > 0,
-            "EIP-3 addresses are not derived in the prover (use ErgoProverBuilder.withEip3Secret)");
+        if (eip3Addresses.isEmpty()) {
+            throw new IllegalStateException("EIP-3 addresses are not derived in the prover (use ErgoProverBuilder.withEip3Secret)");
+        }
```
Comment (aslesarenko) [inline lib-api/src/main/java/org/ergoplatform/appkit/BoxOperations.java]: Why not add checkState to the InternalUtil?

File: lib-impl/src/main/java/org/ergoplatform/appkit/impl/OutBoxBuilderImpl.scala [hunk 3]
```diff
@@ -78,7 +76,9 @@ class OutBoxBuilderImpl(_txB: UnsignedTransactionBuilderImpl) extends OutBoxBuil
   }
 
   override def build: OutBox = {
-    checkState(_contract != null, "Contract is not defined": Any)
+    if (_contract == null) {
+      throw new IllegalStateException("Contract is not defined")
+    }
```
Comment (aslesarenko) [inline lib-impl/src/main/java/org/ergoplatform/appkit/impl/OutBoxBuilderImpl.scala]: Suggest to add method to InternalUtil.

File: common/src/main/java/org/ergoplatform/appkit/Address.java [hunk 4]
```diff
@@ -86,7 +85,7 @@ public NetworkType getNetworkType() {
      * @throws IllegalArgumentException if this instance is not P2PK address
      */
     public P2PKAddress asP2PK() {
-        checkArgument(isP2PK(), "This instance %s is not P2PKAddress", this);
+        InternalUtil.checkArgument(isP2PK(), String.format("This instance %s is not P2PKAddress", this));
```
Comment (aslesarenko) [inline common/src/main/java/org/ergoplatform/appkit/Address.java]: The previous code avoided String.format to be called when no errors.

File: common/src/main/java/org/ergoplatform/appkit/Address.java [hunk 5]
```diff
@@ -100,7 +99,7 @@ public P2PKAddress asP2PK() {
      * @throws IllegalArgumentException if this instance is not P2S address
      */
     public Pay2SAddress asP2S() {
-        checkArgument(isP2S(), "This instance %s is not Pay2SAddress", this);
+        InternalUtil.checkArgument(isP2S(), String.format("This instance %s is not Pay2SAddress", this));
```
Comment (aslesarenko) [inline common/src/main/java/org/ergoplatform/appkit/Address.java]: Same as above.

File: common/src/main/java/org/ergoplatform/appkit/Address.java [hunk 6]
```diff
@@ -279,7 +278,7 @@ public String toString() {
 
     @Override
     public int hashCode() {
-        return Objects.hashCode(_address.hashCode(), _address.networkPrefix());
+        return Objects.hash(_address.hashCode(), _address.networkPrefix());
```
Comment (aslesarenko) [inline common/src/main/java/org/ergoplatform/appkit/Address.java:281]: What is the motivation for this change?

**Review by aslesarenko — CHANGES_REQUESTED** [review]

File: common/src/test/scala/org/ergoplatform/appkit/SecretStorageSpec.scala (hunk 1 above)
Comment (satsen) [reply common/src/test/scala/org/ergoplatform/appkit/SecretStorageSpec.scala:73]: Yes

File: lib-api/src/main/java/org/ergoplatform/appkit/BoxOperations.java (hunk 2 above)
Comment (satsen) [reply lib-api/src/main/java/org/ergoplatform/appkit/BoxOperations.java]: Felt unnecessary for 2 uses, and in my opinion this code is clearer. But I can add it if you want. @aslesarenko

Edit: Added

File: common/src/main/java/org/ergoplatform/appkit/Address.java (hunk 4 above)
Comment (satsen) [reply common/src/main/java/org/ergoplatform/appkit/Address.java:88]: Is it it ok to just use a string concatenation then

File: common/src/main/java/org/ergoplatform/appkit/Address.java (hunk 6 above)
Comment (satsen) [reply common/src/main/java/org/ergoplatform/appkit/Address.java:281]: com.google.common.base.Objects.hashCode is a method in the Guava library.
java.util.Objects.hash is a method in the JDK.

Their implementation is the same. The motivation is that I removed Guava.

**Review by aslesarenko — APPROVED** [review]
